frappe/public/js/frappe/form/grid.js

```diff
@@ -834,7 +834,13 @@ export default class Grid {
 				if (!this.df.data) {
 					this.df.data = this.get_data() || [];
 				}
-				this.df.data.push({ idx: this.df.data.length + 1, __islocal: true });
+				const defaults = this.docfields.reduce((acc, d) => {
+					acc[d.fieldname] = d.default;
+					return acc;
+				}, {});
+				this.df.data.push(
+					$.extend({ idx: this.df.data.length + 1, __islocal: true }, defaults)
+				);
```

Comment:
```suggestion
				this.df.data.push({ idx: this.df.data.length + 1, __islocal: true, ...defaults});
```

The [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals) looks a bit cleaner here and we don't need a library.